exist-core/src/test/java/org/exist/xquery/ContextItemTests.java

```diff
@@ -1,90 +1,100 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
 package org.exist.xquery;
 
-import com.evolvedbinary.j8fu.Either;
-import org.exist.EXistException;
-import org.exist.security.PermissionDeniedException;
-import org.exist.storage.BrokerPool;
-import org.exist.storage.DBBroker;
-import org.exist.test.ExistEmbeddedServer;
-import org.exist.xquery.value.Sequence;
+import org.exist.test.ExistXmldbEmbeddedServer;
```

Comment:
Why is this rewritten to use the XML:DB API when the XMLDB:API is not needed here? I think that adds an extra level of indirection and complexity which is not required. Instead you can talk directly to the database as was the case before.

Reply:
Because that is the code I had at hand, which by the way was merged as part of a previous PR. I am happy to rewrite the test using `org.exist.test.ExistEmbeddedServer` if that makes the tests more stable or faster or both and can still be testing for the static error message.

Reply:
I think you likely only need to revert your change to this class, and add in the additional test that you wanted.

Reply:
I would like to keep it as it is, please see my comment below.